src/Dialect/ONNX/ONNXOps/Math/DFT.cpp

```diff
@@ -74,15 +108,35 @@ LogicalResult ONNXDFTOp::inferShapes(
   if (!hasShapeAndRank(getInput()))
     return success();
 
+  // Not yet shaped, wait for later.
+  if (!isNoneValue(getAxis()) && !hasShapeAndRank(getAxis()))
+    return success();
+
   Type elementType = getInput().getType().cast<ShapedType>().getElementType();
   ONNXDFTOpShapeHelper shapeHelper(getOperation(), {});
   return shapeHelper.computeShapeAndUpdateType(elementType);
 }
 
+//===----------------------------------------------------------------------===//
+// DFT legacy: DFTV17  Shape Inference
+//===----------------------------------------------------------------------===//
+
+LogicalResult ONNXDFTV17Op::inferShapes(
+    std::function<void(mlir::Region &)> doShapeInference) {
+  // Cannot infer the output shape if the input shape is not yet known.
+  if (!hasShapeAndRank(getInput()))
+    return success();
+
+  Type elementType = getInput().getType().cast<ShapedType>().getElementType();
+  ONNXDFTV17OpShapeHelper shapeHelper(getOperation(), {});
+  return shapeHelper.computeShapeAndUpdateType(elementType);
+}
```

Comment:
Since `ONNXDFTV17Op` is decomposed into `ONNXDFTOp`, we don't need to define `inferShape` for it.

Reply:
Okay let me fix it!!